UnitTests/BraintreeTestShared/MockAPIClient.swift

```diff
@@ -17,6 +17,7 @@ public class MockAPIClient: BTAPIClient {
     public var postedIsVaultRequest = false
     public var postedMerchantExperiment: String? = nil
     public var postedPaymentMethodsDisplayed: String? = nil
+    public var postedAppSwitchURL: [String: String?] = [:]
```

Comment:
This is a dictionary because we currently don't have a way to verify the parameters sent in each event. The way the other parameters are handled (for example, `postedIsVaultRequest`) only allows us to evaluate the parameters sent in the last event (in cases where more than one event is sent, like in the `invokedOpenURLSuccessfully` method in my unit test)

Reply:
In my unit test, the events appSwitchFailed and tokenizeFailed are sent, in that order. The second event doesn’t have the appSwitchUrl parameter, which returns nil and prevents me from validating that the events are being sent correctly

Reply:
Ah, I see you are saying the test is to verify that we are sending it for that specific event and not just that it is being set, which makes sense. Though I don't think it would be set as nil, just tested locally and it would still be set if it's being passed for any of the events in `mockAPIClient`. If we are instead trying to verify the URL is sent for this specific event I think what you have makes sense.

Reply:
`appSwitchURL` isn’t sent in all events, and yes, we’re testing that the parameter is sent in the events specified in this ticket